src/core/libTerm.ml

```diff
@@ -186,6 +186,36 @@ module Meta = struct
   let has : bool -> term -> bool =
     let exception Found in fun b t ->
     try iter b (fun _ -> raise Found) t; false with Found -> true
+
+  (** [m_a < m_b] if [m_b] depends on [m_a] transitively, such that the least
+      element has less dependencies. It is the transitive closure of the
+      inverse of [depends]. *)
+  let ( < ) : t -> t -> bool = fun m_a m_b ->
+    (* [depends m_a m_b] is true if metavariable [m_b] occurs in the type of
+       [m_a] ([m_a] depends on [m_b]). *)
+    let depends m_a m_b: bool = occurs m_b !(m_a.meta_type) in
+    depends m_b m_a ||
+    MetaSet.exists (fun m -> depends m m_a) (get true !(m_b.meta_type))
```

Comment:
< doesn't seem transitive as it is not recursive. Moreover, it would be easier to understand to not permute the arguments of < and depends.
```
let rec lt m n = let t = !(n.meta_type) in occurs m t || MetaSet.exists (lt m) (get true t)
```
But it is not very efficient as occurs is defined with iter (false), and get as well. Moreover, we do not need to generate the set of all metas to check dependency. A single iter (true) should be enough.
```
let lt =
  let exception Found in
  fun m ->
  let f n = if n == m then raise Found in
  fun n ->
  try iter true f !(n.meta_type); false with Found -> true
```
In fact this is the same function as occurs but with iter true instead of iter false.
-> you could add a boolean argument to occurs and use occurs true instead.

Reply:
This function is recursive because `get` retrieves all metavariables recursively I think, you may see the unit test below that checks this. And for the order of arguments, the idea is that the minimal argument doesn't depend on anything. Good idea for `occurs`.